### ISSUE TITLE
[New feature suggestions] add history (loss function/accuracy) at each iteration (shapelet model)

### ISSUE THREAD

**Issue by charlatteD** [rtavenar/tslearn#118]
Hi Romain, 

Thank you so much for your amazing input! 

I'm new to the python/deep learning/keras community. 

I'm wondering if the shapelet model could output the history (e.g. loss function/accuracy) at each iteration? Or could you point out the way to output the history if the current version of the code has this output? 

Thank you again! 
Charlotte

**Comment by rtavenar** [comment]
Hi Charlotte,

Thank you for this feedback, and for this suggestion. I will check asap whether this could be implemented easily or not, and let you know.

**Comment by rtavenar** [comment]
Hi @charlatteD 

After a quick check, this is available, but a bit hidden at the moment:

```
clf = ShapeletModel(n_shapelets_per_size={10: 3},
                    optimizer=Adagrad(lr=.01),
                    weight_regularizer=0.1,
                    max_iter=1000,
                    verbose_level=2,
                    random_state=None)

clf.fit(X_train, y_train)
print(clf.model.history.history)
```

It could be a good idea to update the example notebook in the doc so that people are aware of that possibility. I will tag that as a good first issue if someone is interested to update the docs, or do it later (but not sure when).

Let me know if this fits your needs.

**Comment by charlatteD** [comment]
Hi @rtavenar 

Yes, it's exactly what I want. Thank you so much! I've been testing your shapelet code on my project for quite a while. But I'm very new to the github community. Maybe I will update the docs once I'm better at navigating through the github system. Thanks!!

Best,
Charlotte

**Comment by rtavenar** [comment]
Great!

So for anyone willing to add this info to the docs, one simple idea would be to add a plot of training accuracy in this example:
https://github.com/rtavenar/tslearn/blob/master/tslearn/docs/examples/plot_shapelets.py

**Comment by rtavenar** [comment]
Will appear in the doc once version 0.2 is released
